ui/src/overview.tsx

```diff
@@ -117,9 +120,54 @@ function AddCriteriaButton(props: { group: string | GroupKind }) {
   );
 }
 
-function ParticipantsGroup(props: { group: Group }) {
+function ParticipantsGroup(props: { group: Group; index: number }) {
+  const dispatch = useAppDispatch();
+  const cohort = useCohort();
+  const [renameGroupDialog, showRenameGroup] = useTextInputDialog({
+    title: "Edit Group Name",
+    initialText:
+      props.group.name === ""
+        ? "Group " + String(props.index + 1)
+        : props.group.name,
+    textLabel: "Group Name",
+    buttonLabel: "Confirm",
```

Comment:
`"Rename Group"`

The UI is easier to understand if we use more specific wording. That's why we added this as a parameter in the first place.

Reply:
done